libr/core/cmd_search_rop.c

```diff
@@ -52,9 +52,101 @@ static RList* get_constants (const char *str) {
 	return list;
 }
 
+static bool isFlag (const char *f) {
+	const char *flags = "zf,pf,sf,cf,of";
+
+	if (strstr (flags, f)) {
+		return true;
+	} else {
+		return false;
+	}
+}
+
+// TODO: hideous code, must rafactor
+// binary op
+static ut64 simulate_op (const char *op, ut64 src1, ut64 src2, ut64 old_src1, ut64 old_src2) {
+
+	if (!strcmp (op, "^")) {
+		return src1 ^ src2;
+	} else if (!strcmp (op, "+")) {
+		return src1 + src2;
+	} else if (!strcmp (op, "-")) {
+		return src1 - src2;
+	} else if (!strcmp (op, "*")) {
+		return src1 * src2;
+	} else if (!strcmp (op, "/")) {
+		return src1 / src2;
+	} else if (!strcmp (op, "%")) {
+		return src1 % src2;
+	} else if (!strcmp (op, "<<")) {
+		return src1 << src2;
+	} else if (!strcmp (op, ">>")) {
+		return src1 >> src2;
+	} else if (!strcmp (op, "&")) {
+		return src1 & src2;
+	} else if (!strcmp (op, "+=")) {
+		return old_src1 + src2;
+	} else if (!strcmp (op, "-=")) {
+		return old_src1 - src2;
+	} else if (!strcmp (op, "*=")) {
+		return old_src1 * src2;
+	} else if (!strcmp (op, "/=")) {
+		return old_src1 / src2;
+	} else if (!strcmp (op, "%=")) {
+		return old_src1 % src2;
+	} else if (!strcmp (op, "<<")) {
+		return src1 << src2;
+	} else if (!strcmp (op, ">>")) {
+		return src1 >> src2;
+	} else if (!strcmp (op, "&=")) {
+		return src1 & src2;
+	} else if (!strcmp (op, "^=")) {
+		return src1 ^ src2;
+	} else if (!strcmp (op, "|=")) {
+		return src1 | src2;
+	}
+
+	// fail value
+	return -1337;
```

Comment:
dont mix error and data channels. return the value as a ref and return bool for example